src/util/fs.js

```diff
@@ -6,34 +6,47 @@ import * as promise from './promise.js';
 import {promisify} from './promise.js';
 import map from './map.js';
 
+const userHome = require('./user-home-dir').default;
 const fs = require('fs');
 const globModule = require('glob');
 const os = require('os');
 const path = require('path');
 
 export const lockQueue = new BlockingQueue('fs lock');
 
-export const readFileBuffer = promisify(fs.readFile);
-export const writeFile: (path: string, data: string) => Promise<void> = promisify(fs.writeFile);
-export const readlink: (path: string, opts: void) => Promise<string> = promisify(fs.readlink);
-export const realpath: (path: string, opts: void) => Promise<string> = promisify(fs.realpath);
-export const readdir: (path: string, opts: void) => Promise<Array<string>> = promisify(fs.readdir);
-export const rename: (oldPath: string, newPath: string) => Promise<void> = promisify(fs.rename);
-export const access: (path: string, mode?: number) => Promise<void> = promisify(fs.access);
-export const stat: (path: string) => Promise<fs.Stats> = promisify(fs.stat);
-export const unlink: (path: string) => Promise<void> = promisify(require('rimraf'));
-export const mkdirp: (path: string) => Promise<void> = promisify(require('mkdirp'));
-export const exists: (path: string) => Promise<boolean> = promisify(fs.exists, true);
-export const lstat: (path: string) => Promise<fs.Stats> = promisify(fs.lstat);
-export const chmod: (path: string, mode: number | string) => Promise<void> = promisify(fs.chmod);
-export const link: (path: string) => Promise<fs.Stats> = promisify(fs.link);
-export const glob: (path: string) => Promise<Array<string>> = promisify(globModule);
+export const createWriteStream = (path: string, options?: Object): fs.WriteStream =>
+  fs.createWriteStream(expandPath(path), options);
```

Comment:
Do we need to expandPath in every fs method?
I would add it only in the source of config.getOption method instead.
I don't think `~` needs to be expanded in every fs operation